docs/v2_design.md

```diff
@@ -0,0 +1,189 @@
+
+static_assert(std::is_same_v<decltype(p), quantity<reference<isq::power_dim, si::watt>{}, int>>);
+```
+
+### Named quantity/dimension and derived (unnamed) unit
+
+```cpp
+auto s = 5 * si::speed[m / s];
+
+static_assert(std::is_same_v<decltype(s),
+              quantity<reference<isq::speed_dim, derived_unit<si::metre, per<si::second>>>{}, int>>);
+```
+
+### Derived (unnamed) quantity/dimension and derived (unnamed) unit
+
+```cpp
+auto q = 10 * si::length[m] / (2 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q),
+              quantity<reference<derived_dimension<isq::length_dim, per<isq::time_dim>>,
+                       derived_unit<si::metre, per<si::second>>>{}, int>>);
+```
+
+### Base quantity as a result of dimensional transformation
+
+```cpp
+auto d = 5 * si::speed[m / s] * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(d), quantity<reference<isq::length_dim, si::metre>{}, int>>);
+```
+
+### Dimensionless
+
+```cpp
+auto q = 20 * si::speed[m / s] / (10 * si::length[m]) * (5 * si::time[s]);
+
+static_assert(std::is_same_v<decltype(q), quantity<reference<one_dim, one>{}, int>>);
+```
+
+---
+
+### Quantity Comparison
+
+```cpp
+// Same dimension type & different unit
+static_assert(1000 * si::length[m] == 1 * si::length[km]);
+
+// Named and derived dimensions (same units)
+static_assert(10 * si::length[m] / (2 * si::time[s]) == 5 * si::speed[m / s]);
+
+// Same named dimension & different but equivalent unit
+static_assert(10 * si::frequency[1 / s] == 10 * si::frequency[Hz]);
+
+// Named and derived dimensions (different but equivalent units)
+static_assert(10 / (2 * si::time[s]) == 5 * si::frequency[Hz]);
+
+// Different named dimensions
+// static_assert(5 * si::activity[Bq] == 5 * si::frequency[Hz]); // ambiguous
+```
+
+---
+
+## Quantity Arithmetics
+
+```cpp
+// Named and derived dimensions (same units)
+auto q1 = 10 * si::length[m] / (2 * si::time[s]) + 5 * si::speed[m / s];
+static_assert(is_same_v<decltype(q1),
+              quantity<reference<isq::speed_dim, derived_unit<si::metre, per<si::second>>>{}, int>>)
+
+// Named and derived dimensions (different units)
+auto q2 = 10 / (2 * si::time[s]) + 5 * si::frequency[Hz];
+static_assert(is_same_v<decltype(q2), quantity<reference<isq::frequency_dim, si::hertz>{}, int>>)
+
+// Different named dimensions
+// auto q3 = 5 * si::activity[Bq] + 5 * si::frequency[Hz];                          // ambiguous
+// auto q4 = 5 * si::activity[Bq] + 10 / (2 * si::time[s]) + 5 * si::frequency[Hz]; // ambiguous
+```
+
+---
+
+## Dimensions
+
+```cpp
+// If something compares equal it is always equivalent and convertible
+static_assert(isq::speed_dim == isq::speed_dim);
+static_assert(equivalent(isq::speed_dim, isq::speed_dim));
+static_assert(convertible(isq::speed_dim, isq::speed_dim));
+
+// Derived unnamed vs named
+// Should the `equivalent` be consistent with the `op=`?
+static_assert(isq::length_dim / isq::time_dim != isq::speed_dim);              // ???
```

Comment:
Does it make sense to add also support for `+` and `-` on dimensions and units?
For example:

```cpp
isq::length_dim / isq::time_dim + isq::speed_dim == isq::speed_dim;
```